eventhandlers/action_runners/actions/VertexTrainDatasetAction.py

```diff
@@ -0,0 +1,291 @@
+class VertexTrainDatasetAction(ActionRunner):
+    public_name = 'Vertex AI Train Dataset (Google / GCP)'
+    description = 'Train model with Vertex AI'
+    icon = 'https://www.svgrepo.com/show/375510/vertexai.svg'
+    precondition = ActionCondition(default_event = None, event_list = [])
+    kind = 'VertexTrainDatasetAction'  # The kind has to be unique to all actions
+    category = 'training'  # Optional
+
+    # What events can this action listen to?
+    trigger_data = ActionTrigger(default_event = 'manual_trigger',
+                            event_list = ['manual_trigger'])
+
+    # What pre-conditions can this action have?
+    condition_data = ActionCondition(default_event = 'some_diffgram_event',
+                                     event_list = ['some_diffgram_event'])
+
+    # How to declare the actions as completed?
+    completion_condition_data = ActionCompleteCondition(default_event = 'action_completed',
+                                                        event_list = ['action_completed'])
+
+    def execute_pre_conditions(self, session) -> bool:
+        run = ActionRun.get_latest_action_status(self.session, self.action.id)
+        if run != None and run.status == 'running':
+            return False
+        return True
+
+    def execute_action(self, session):
+        ActionRun.set_action_run_status(self.session, self.action_run.id, "running")
+        temp_folder_name = f"temp_{self.action_run.id}"
+
+        connection_strategy = ConnectionStrategy(
+            connection_class = GoogleCloudStorageConnector,
+            connection_id = self.action.config_data.get('connection_id'),
+            session = self.session
+        )
+
+        [google_vertex_connector, success] = connection_strategy.get_connector()
+
+        if not success:
+            ActionRun.set_action_run_status(self.session, self.action_run.id, "failed")
+            return
+        
+        credentials = google_vertex_connector.get_credentials()
+        
+        self.init_ai_platform(google_vertex_connector)
+        gcp_data_tools = self.init_gcp_data_tools(google_vertex_connector)
+
+        file_list = self.get_file_list(session)
+
+        dataset_file_list = []
+        for file in file_list:
+            if file.image is not None:
+                file_link = self.write_diffgram_blob_to_gcp(gcp_data_tools, temp_folder_name, file)
+                dataset_file_list.append({
+                    "filename": f"gs://{self.action.config_data.get('staging_bucket_name_without_gs_prefix')}/{file_link}",
+                    "diffgram_file": file
+                })
+
+        self.count_dataset_labels(session, dataset_file_list)
+        vertexai_import_file = self.write_vertex_format_jsonl_file(gcp_data_tools, temp_folder_name, dataset_file_list)
+
+        self.clean_up_temp_dir(temp_folder_name)
+
+        datasets_list = aiplatform.datasets.ImageDataset.list()
+        existing_datasets = []
+
+        for dataset in datasets_list:
```

Comment:
We can maybe move this to a function separately with a appropriate name to understand what this for loop does.